pkg/provisioner/cloudprovisioner_azuredisk.go

```diff
@@ -0,0 +1,936 @@
+/*
```

Comment:
We can just call this file `cloudprovisioner.go`. Thinking some more about naming, I window if we should call this object `AzureCloudProvisioner` or even more simply `AzureProvisioner`? I am thinking it makes sense to integrate the CRD-based volume management into the driver by implementing a `AzureDiskCrdProvisioner` that creates/updates/deletes the CRDs instead of calling into the cloud provisioner.